client/src/sagas/watchFetchMovies.js

```diff
@@ -0,0 +1,42 @@
+import {put, takeEvery} from 'redux-saga/effects';
+
+import SEARCH_FILTER_TYPE from '../components/SearchFilter/searchFilterType'
+import SORT_TYPE from '../components/Sort/sortType';
+import {MOVIES_FETCH_REQUESTED, MOVIES_FETCH_SUCCEEDED, MOVIES_FETCH_FAILED} from '../actions/actionTypes';
+
+const getFilterPart = (filterType, filterValue) => {
+    if (filterValue==="") return "";
+    switch (filterType) {
+        case SEARCH_FILTER_TYPE.title:
+            return "&search="+filterValue+"&searchBy=title";
+        case SEARCH_FILTER_TYPE.genre:
+            return "&search="+filterValue+"&searchBy=genres"
+    };
+};
+
+const getSortPart = (sortType) => {
+    switch (sortType) {
+        case SORT_TYPE.releaseDate:
+            return "&sortBy=release_date&sortOrder=desc";
+        case SORT_TYPE.rating:
+            return "&sortBy=vote_average&sortOrder=desc"
+    }
+};
+
+
+function* fetchMovies(action) {
+    try {
+        const movies = yield fetch('https://reactjs-cdp.herokuapp.com/movies?limit=5000' + getFilterPart(action.payload.filterType, action.payload.filterValue) + getSortPart(action.payload.sortType)).then(function (response) {
+            return response.json();
```

Comment:
_minor_
I wonder if this returned value is used further

Reply:
Well, because of the line below:
`yield put({ type: MOVIES_FETCH_SUCCEEDED, movies });`
this returned value is used in [movies reducer](https://github.com/AllaShtatska/react-mentoring-program/blob/5-flux-redux/client/src/reducers/movies.js):
`case MOVIES_FETCH_SUCCEEDED:
            return {
                ...state,
                list: action.movies.data
            };`

Reply:
Oh, I'm sorry.
I've been confused a little, probably it would worth to split a long line a little bit and don't mix `then` with yield promise technique?

```javascript
function* fetchMovies(action) {
  const baseAPIUrl = 'https://reactjs-cdp.herokuapp.com/movies?limit=5000';
  const filterQuery = getFilterPart(action.payload.filterType, action.payload.filterValue);
  const sortQuery = getSortPart(action.payload.sortType);
  const apiURL = `${baseAPIUrl}${filterQuery}${sortQuery}`;

    try {
        const response = yield fetch(apiURL);
        const movies = yield response.json();

        yield put({ type: MOVIES_FETCH_SUCCEEDED, movies });
    } catch (error) {
        yield put({ type: MOVIES_FETCH_FAILED, error });
    }
};
```